### PR TITLE
chore: Deletes outdated integration tests

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Deletes outdated integration tests. They are not being run or used.

Deleted folders:
- examples/test-upgrade
- integrationtesting

Link to any related issue(s): CLOUDP-236583

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: website/docs/r/cloud_backup_snapshot_restore_job.html.markdown [hunk 1]
```diff
@@ -126,9 +126,7 @@ resource "mongodbatlas_cloud_backup_snapshot_restore_job" "test" {
 ```
 
 ### Available complete examples
-- [Restore from automated backup snapshot](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v110/cloud-backup-snapshot/v110)
-- [Restore from backup snapshot download](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/test-upgrade/v100/design-id-reference/snapshot-restore)
```
Comment (maastha) [inline website/docs/r/cloud_backup_snapshot_restore_job.html.markdown:130]: are we sure these are outdated? If these are relevant use-cases, could we look into just keeping these as examples?

Comment (lantoli) [reply website/docs/r/cloud_backup_snapshot_restore_job.html.markdown:130]: they are 3 links,  link 2 and 3 are broken, I updated link 3.
checking if link 1 is relevant and will move it to examples folder or will use an existing one

Comment (lantoli) [reply website/docs/r/cloud_backup_snapshot_restore_job.html.markdown:130]: updated to have 2 meaningful examples

**Review by maastha — APPROVED** [review]
LGTM, thanks for addressing the comments!

**Pull request merged**